File: android/app/src/main/java/com/izzatismail/midtrim/domain/usecase/GenerateThumbnailUseCase.kt
package com.izzatismail.midtrim.domain.usecase

import com.izzatismail.midtrim.domain.repository.FrameExtractor

class GenerateThumbnailUseCase(
    private val frameExtractor: FrameExtractor
) {
    suspend fun execute(videoUri: String): String {
        return frameExtractor.extractFrame(videoUri, atTime = 0.0)
    }
}

File: ios/MidTrim/MidTrim/Domain/UseCases/GenerateThumbnailUseCase.swift
import Foundation

struct GenerateThumbnailUseCase {
    private let frameExtractor: FrameExtractorProtocol

    init(frameExtractor: FrameExtractorProtocol) {
        self.frameExtractor = frameExtractor
    }

    func execute(from outputVideoURI: String) async throws -> String {
        try await frameExtractor.extractFrame(from: outputVideoURI, at: 0)
    }
}
